testutil/wait.go

```diff
@@ -112,3 +113,27 @@ func (h *height) update(height uint64) {
 	}
 	h.current = height
 }
+
+// WaitForCondition periodically executes the given function fn based on the provided pollingInterval.
+// The function fn should return true of the desired condition is met. If the function never returns true within the timeoutAfter
+// period, or fn returns an error, the condition will not have been met.
+func WaitForCondition(timeoutAfter, pollingInterval time.Duration, fn func() (bool, error)) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutAfter)
+	defer cancel()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("failed waiting for condition after %f seconds", timeoutAfter.Seconds())
+		case <-time.After(pollingInterval):
```

Comment:
Fyi, there is a resource leak here if `<-ctx.Done()` happens first. 

Per https://pkg.go.dev/time#After

> The underlying Timer is not recovered by the garbage collector until the timer fires. If efficiency is a concern, use NewTimer instead and call Timer.Stop if the timer is no longer needed.

I'm not sure if efficiency is a concern here. But to be safe, I typically `NewTimer` and `defer timer.Stop()` to be safe.